src/assets.rs

```diff
@@ -284,6 +303,42 @@ impl HighlightingAssets {
     }
 }
 
+/// A SyntaxSet in serialized form, i.e. bincoded and flate2 compressed.
+/// We keep it in this format since we want to load it lazily.
+#[derive(Debug)]
+enum SerializedSyntaxSet {
+    /// The data comes from a user-generated cache file.
+    FromFile(PathBuf),
+
+    /// The data to use is embedded into the bat binary.
+    FromBinary(&'static [u8]),
+}
+
+impl SerializedSyntaxSet {
+    fn deserialize(&self) -> SyntaxSet {
+        match self {
+            SerializedSyntaxSet::FromBinary(data) => {
+                from_binary(data)
+            },
+            SerializedSyntaxSet::FromFile(ref path) => {
+                asset_from_cache(&path, "syntax set").expect("cache corrupt, consider rebuilding or clearing, see https://github.com/sharkdp/bat#adding-new-syntaxes--language-definitions on how")
```

Comment:
One downside of lazy loading is that we can't fail "nicely" early on. We could return a Result here and propagate it onwards, but it would cascade into very unpleasant code since then get_syntax_set() would also need to return a Result. It is enough of an edge for this to fail to justify .expect() here, IMHO.

Reply:
Sounds okay to me for now.